static/js/settings_org.js

```diff
@@ -422,46 +462,38 @@ function _set_up() {
             }
         });
 
-        var change_process_button = subsection.find('.subsection-header .button');
-        change_process_button.first().text(i18n.t("Save"));
-        if (show_change_process_button) {
-            change_process_button.removeClass('hide').addClass('show');
-        } else {
-            change_process_button.removeClass('show').addClass('hide');
-        }
+        var save_btn_controls = subsection.find('.subsection-header .save-button-controls');
+        var button_state = (show_change_process_button) ? "unsaved" : "saved";
+        exports.change_save_button_state(save_btn_controls, button_state);
     });
 
-    $('.organization').on('click', '.subsection-header .subsection-changes-discard button', function (e) {
+    $('.organization').on('click', '.subsection-header .subsection-changes-discard .button', function (e) {
         e.preventDefault();
         e.stopPropagation();
         _.each(get_subsection_property_elements(e.target), discard_property_element_changes);
-        var subsection = $(e.target).closest('.org-subsection-parent');
-        var change_process_buttons = subsection.find('.subsection-header .button');
-        change_process_buttons.removeClass('show').addClass('hide');
+        var save_btn_controls = $(e.target).closest('.save-button-controls');
+        exports.change_save_button_state(save_btn_controls, "discarded");
     });
 
     exports.save_organization_settings = function (data, save_button, success_continuation) {
         var subsection_parent = save_button.closest('.org-subsection-parent');
-        var discard_button = subsection_parent.find('.subsection-changes-discard button');
+        var save_btn_container = subsection_parent.find('.save-button-controls');
         var failed_alert_elem = subsection_parent.prevAll('.admin-realm-failed-change-status:first').expectOne();
-        save_button.text(i18n.t("Saving"));
-        save_button.attr("data-status", "saving");
+        exports.change_save_button_state(save_btn_container, "saving");
         channel.patch({
             url: "/json/realm",
             data: data,
             success: function (response_data) {
-                discard_button.removeClass('show').addClass('hide');
                 failed_alert_elem.hide();
-                save_button.attr("data-status", "saved");
-                save_button.text(i18n.t("Saved"));
-                save_button.removeClass('hide').addClass('show').stop(true).fadeTo(0, 1);
+                setTimeout(function () {
+                  exports.change_save_button_state(save_btn_container, "succeeded");
+                }, 500);
```

Comment:
Is there a reason you didn't put the `setTimeout` inside `export.change_save_button_state`?

Reply:
I mostly kept the setTimeout outside the change_save_button_state for testability and keeping the function tight. This way it can be used synchronously should the need arise. I imagine helper functions like this being shared across different sections/modules once things become a little more consistent in combination with the associated template.